### ISSUE TITLE
[a11y]: MultiSelect selected items do not need a gray fill when it is redundant

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon#13120]
Original guidance from @mbgower in https://github.com/carbon-design-system/carbon/issues/12723:

> ### Issue 2 Highlighting should only be on items with focus, not on selected items
> In the multi-select, items that are selected are both checked AND have a fill added. This is needlessly redundant AND causes difficulties navigating since the same fill is added to show the item with focus. A user cannot see _any_ evidence of focus moving when moving through multiple selected items. This is shown in the following screen snippet where the item with focus is option 2. This may be resolved if the issue on increasing contrast on the item with focus, https://github.com/carbon-design-system/carbon/issues/12720,  is addressed.
> ![image](https://user-images.githubusercontent.com/14298245/203652771-c4d6ba79-b73b-44af-b9f7-f53225030d52.png)

We're considering this for the future, but it has a large design impact and will need exploration.

**Comment by pratikkarad** [comment]
Hi, I would like to try my hands at this. Please assign it to me.

**Comment by tw15egan** [comment]
This was fixed via https://github.com/carbon-design-system/carbon/pull/14276
